cpp/src/arrow/flight/sql/odbc/odbcabstraction/odbc_impl/odbc_statement.cc

```diff
@@ -583,8 +583,8 @@ void ODBCStatement::SetStmtAttr(SQLINTEGER statementAttribute, SQLPOINTER value,
     case SQL_ATTR_ASYNC_ENABLE:
 #ifdef SQL_ATTR_ASYNC_STMT_EVENT
     case SQL_ATTR_ASYNC_STMT_EVENT:
-      throw DriverException("Unsupported attribute", "HYC00");
 #endif
+      throw DriverException("Unsupported attribute", "HYC00");
```

Comment:
Want to just add this to SQL_ATTR_ASYNC_ENABLE explicitly rather than fall-through? Might be easier to read.

Reply:
Sure, fixed